spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitListenerObservation.java

```diff
@@ -86,8 +105,13 @@ public static class DefaultRabbitListenerObservationConvention implements Rabbit
 
 		@Override
 		public KeyValues getLowCardinalityKeyValues(RabbitMessageReceiverContext context) {
-			return KeyValues.of(RabbitListenerObservation.ListenerLowCardinalityTags.LISTENER_ID.asString(),
-							context.getListenerId());
+			return KeyValues.of(
+					RabbitListenerObservation.ListenerLowCardinalityTags.LISTENER_ID.asString(), context.getListenerId(),
+					RabbitListenerObservation.ListenerLowCardinalityTags.EXCHANGE.asString(),
+					context.getCarrier().getMessageProperties().getReceivedExchange(),
+					RabbitListenerObservation.ListenerLowCardinalityTags.ROUTING_KEY.asString(),
+					context.getCarrier().getMessageProperties().getReceivedRoutingKey()
```

Comment:
I think for slight optimization the `context.getCarrier().getMessageProperties()` could be extracted into a local variable.

Reply:
done